### PR TITLE
fix: actions using `node12 which is deprecated`

### LINKED ISSUE #424: .github/workflows/tests.yml `uses node12 which is deprecated`
https://github.com/mixpanel/mixpanel-js/actions/runs/8886009271

### PR DESCRIPTION
**Pull request by hamirmahal**
closes #424.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -2,24 +2,23 @@ name: Tests
...
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm test
+      - uses: actions/checkout@v4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
```
Comment (hamirmahal) [inline .github/workflows/tests.yml:20]: These are the latest versions currently available.

**Comment by hans-lizihan** [comment]
thank you!

**Review by hans-lizihan — APPROVED** [review]

**Comment by hamirmahal** [comment]
You're welcome!

**Comment by hamirmahal** [comment]
You're welcome!

**Pull request merged**